legal-api/migrations/versions/f1d010259785_modify_data_type_for_max_shares.py

```diff
@@ -0,0 +1,43 @@
+"""modify_data_type_for_max_shares
+
+Revision ID: f1d010259785
+Revises: ad21c1ed551e
+Create Date: 2025-02-28 22:29:38.543965
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'f1d010259785'
+down_revision = 'ad21c1ed551e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute('ALTER TABLE share_classes ALTER COLUMN max_shares TYPE NUMERIC(20) USING max_shares::NUMERIC(20);')
+    op.execute('ALTER TABLE share_series ALTER COLUMN max_shares TYPE NUMERIC(20) USING max_shares::NUMERIC(20);')
+
+    op.execute('ALTER TABLE share_classes_version ALTER COLUMN max_shares TYPE NUMERIC(20) USING max_shares::NUMERIC(20);')
+    op.execute('ALTER TABLE share_series_version ALTER COLUMN max_shares TYPE NUMERIC(20) USING max_shares::NUMERIC(20);')
+
+
+def downgrade():
+    op.execute("UPDATE share_classes SET max_shares = 2147483647 WHERE max_shares > 2147483647;")
+    op.execute("UPDATE share_classes SET max_shares = -2147483648 WHERE max_shares < -2147483648;")
```

Comment:
`max_shares` should not have any negative value in any scenario except the bad data, so just wondering if I need to check it and if I can update the negative number to 0 by default.

Reply:
i feel like we should allow the bad value.  but we should ask David R tomorrow at standup

Reply:
i misunderstood this question.  i think we can leave your implementation as is wrt to the negative value scenario.  It shouldn't happen  but if it does I think leaving it as a negative value is fine